OneCommunityGlobal/HighestGoodNetworkApp
# Induja - Update DropBox Link and text on Weekly Summaries Reports page

# Description
Update DropBox Link and text on Weekly Summaries Reports page 
Any user Login → Reports → Weekly Summaries Reports 
Despite a warning telling people not to, many keep editing their DropBox link when submitting their summaries. The links they change to don’t work in many cases because they don’t know how to create a proper DB share link. I want to remove the ability for people to edit this link here.
Change everything about the link to a clickable “Your DropBox Media Files Link (Share your files here)” that will be the link they are set up with during New User Setup and that is only editable from their Profile Page
Remove the “(required)” from the other point… all of these are required and having that there implies otherwise
<img width="747" alt="Screenshot 2023-08-11 at 2 50 49 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/11251850/df3a5a86-7369-4a79-ad44-e86276470f34">


Fixes (Priority High)


## Main changes explained:
1. Users can input a Dropbox link only if the link field is empty.
2. The Dropbox link can be added only once on the summary page. It is saved upon adding. Users can add or modify the link through the profile page.
3. If a Dropbox link is added, the media label becomes clickable, and clicking it redirects to the linked media folder.
4. Once a Dropbox media link is added and saved, it cannot be modified from the Weekly Summary Update page. Modifications can only be made from the profile page.

## How to test:

1. check into current branch
2. do npm run start:local to run this PR locally
3. log as any user
4. On the dashboard → Open Weekly Summary
5. Verify and Test Media Link:
      a. If a media link is added during profile setup, ensure it's visible on the dashboard.
      b. Clicking the media link should redirect to the associated media (Dropbox) folder.
6.  Set Up Media Link (if absent):
      a. If no media link is set up, input the media link in the box
      b. The modified link will be saved and used for subsequent submissions.
8. Once the summary is updated and successfully saved, go to Reports-> Weekly Summaries Report
9. Verify if summary is updated accordingly.

## Screenshots or videos of changes:
Case 1:  If Dropbox media link is set up earlier:


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/11251850/ff78b093-2dac-4635-b4e7-21df2451c62b


Case 2:  If Dropbox media link is not set up initially:

If dropbox media link is not setup, user can add the link and verify if its saved for the next update.



https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/11251850/68885bab-ec73-452b-b572-03004f94198e
